### PR TITLE
✨Public and restricted doc editable

### PR DESCRIPTION
**Pull request by AntoLC**
## Purpose

We increased the visibility options for docs:
- public
- restricted
- authenticated

We also have 2 editability options:
- readonly
- editable

The editability options are only available for public and authenticated docs.

## Proposal

- [x] ♻️(frontend) minor components update
- [x] ✨(frontend) increase doc visibility options
- [x] tests

## Demo

[scrnli_VZHQd70DS9qxw9.webm](https://github.com/user-attachments/assets/48fb17ca-933c-4ccc-8b23-186bf8240db4)

### CODE REVIEW

File: src/frontend/apps/impress/src/features/docs/doc-management/types.tsx [hunk 1]
```diff
@@ -27,14 +27,19 @@ export enum LinkReach {
...
 export interface Doc {
   id: string;
   title: string;
   content: Base64;
   link_reach: LinkReach;
-  link_role: 'reader' | 'editor';
+  link_role: LinkRole;
   accesses: Access[];
   created_at: string;
   updated_at: string;
```
Comment (PanchoutNathan) [inline src/frontend/apps/impress/src/features/docs/doc-management/types.tsx:45]: I think this type is not only related to this feature, but can be used everywhere. I moved this file to the root of the feature/docs folder. But maybe in another PR? and maybe. And maybe you don't agree with that

File: src/frontend/apps/impress/src/features/docs/doc-management/components/DocVisibility.tsx [hunk 2]
```diff
@@ -44,55 +58,83 @@ export const DocVisibility = ({ doc }: DocVisibilityProps) => {
...
+              {[
+                {
+                  label: t('Read only'),
+                  value: LinkRole.READER,
+                },
+                {
+                  label: t('Can read and edit'),
+                  value: LinkRole.EDITOR,
+                },
+              ].map((radio) => (
```
Comment (PanchoutNathan) [inline src/frontend/apps/impress/src/features/docs/doc-management/components/DocVisibility.tsx:120]: Maybe create a constant above?

File: src/frontend/apps/impress/src/features/docs/doc-management/components/DocVisibility.tsx [hunk 3]
```diff
@@ -44,55 +58,83 @@ export const DocVisibility = ({ doc }: DocVisibilityProps) => {
...
-                });
-              });
-          }}
-          color="primary"
-          icon={<span className="material-icons">copy</span>}
-        >
-          {t('Copy link')}
-        </Button>
+        {(doc.link_reach === LinkReach.AUTHENTICATED ||
+          doc.link_reach === LinkReach.PUBLIC) && (
```
Comment (PanchoutNathan) [inline src/frontend/apps/impress/src/features/docs/doc-management/components/DocVisibility.tsx:96]: what do you think of creating a constant with explicit naming? like for example : 

`const showRestrictedOptions = doc.link_reach === LinkReach.AUTHENTICATED || doc.link_reach === LinkReach.PUBLIC`

or 

`const showRestrictedOptions = [LinkReach.AUTHENTICATED, LinkReach.PUBLIC].includes(doc.link_reach);`

File: src/frontend/apps/e2e/__tests__/app-impress/doc-member-list.spec.ts [hunk 4]
```diff
@@ -164,14 +164,22 @@ test.describe('Document list members', () => {
     const shareModal = page.getByLabel('Share modal');
```
Comment (PanchoutNathan) [inline src/frontend/apps/e2e/__tests__/app-impress/doc-member-list.spec.ts:164]: Add tests to know if restrictions should be shown or not based on visibility

**Review by AntoLC — COMMENTED** [review]
> ✅ Nice work ! This is a cool feature 

Yes this feature is waited since the beginning of the project. ^^

**Review by PanchoutNathan — APPROVED** [review]

**Pull request merged**